### PR TITLE
Add Syft

### PR DESCRIPTION
**Pull request by ibarakaiev**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/syft/meta.yaml
Comment (JanJaguschQC) [inline recipes/syft/meta.yaml:49]: I'm willing to be listed as a a maintainer.

Comment (janjagusch) [inline recipes/syft/meta.yaml:49]: I'm willing to be listed as a maintainer.

File: recipes/syft/meta.yaml
Comment (xhochy) [inline recipes/syft/meta.yaml:21]:
```suggestion
    - go build -ldflags -v -o $PREFIX/bin/syft  # [not win]
    - go build -ldflags -v -o %LIBRARY_BIN%\syft.exe  # [win]
```

File: recipes/syft/meta.yaml
Comment (xhochy) [inline recipes/syft/meta.yaml:30]:
```suggestion
    - {{ compiler('go_nocgo') }}
```

File: recipes/syft/meta.yaml
Comment (xhochy) [inline recipes/syft/meta.yaml:5]:
```suggestion
```

File: recipes/syft/meta.yaml
Comment (xhochy) [inline recipes/syft/meta.yaml:1]:
```suggestion
{% set goname = "syft" %}
```

File: recipes/syft/conda_build_config.yaml
Comment (xhochy) [inline recipes/syft/conda_build_config.yaml:1]: Please delete this file, the default should suffice if you use the different macro below

File: recipes/syft/meta.yaml
Comment (xhochy) [inline recipes/syft/meta.yaml:12]:
```suggestion
  - folder: src
```

File: recipes/syft/meta.yaml
Comment (xhochy) [inline recipes/syft/meta.yaml:13]:
```suggestion
    url: https://github.com/anchore/syft/archive/v{{ version }}.tar.gz
```

File: recipes/syft/meta.yaml
Comment (xhochy) [inline recipes/syft/meta.yaml:19]:
```suggestion
    - pushd src
```

File: recipes/syft/meta.yaml
Comment (xhochy) [inline recipes/syft/meta.yaml:24]:
```suggestion
    - go-licenses save . --save_path ../library_licenses
```

File: recipes/syft/meta.yaml
Comment (xhochy) [inline recipes/syft/meta.yaml:42]:
```suggestion
    - src/LICENSE
```

File: recipes/syft/meta.yaml
Comment (xhochy) [inline recipes/syft/meta.yaml:38]:
```suggestion
  home: https://github.com/anchore/syft
```

**Review by xhochy — CHANGES_REQUESTED** [review]
The template I gave you did still contain too much cruft. 🙈

**Comment by xhochy** [comment]
```
go: errors parsing go.mod:
/home/conda/staged-recipes/build_artifacts/syft_1695976776926/work/src/go.mod:3: invalid go version '1.21.0': must match format 1.23
```

Please fix this because requesting review.

**Comment by janjagusch** [comment]
> ```
> go: errors parsing go.mod:
> /home/conda/staged-recipes/build_artifacts/syft_1695976776926/work/src/go.mod:3: invalid go version '1.21.0': must match format 1.23
> ```
> 
> Please fix this because requesting review.

@xhochy, sorry for the lack of context when requesting your review.

We're trying to figure out how to deal with the error you've mentioned. We have some questions that maybe you could answer:
- `syft` requires Go 1.21.0([source](https://github.com/anchore/syft/blob/1067dc2ce9304c8c5b094a6e1b4ad3beabb91746/go.mod#L3)), but that version is not available on CF yet ([source](https://prefix.dev/channels/conda-forge/packages/go-nocgo)). How should we proceed here? Wait for a new Go release on CF? Patch the `go.mod` file? Package an older version, for example, v0.89.0 ([source](https://github.com/anchore/syft/blob/v0.89.0/go.mod))?
- However, the parsing error suggests that we should match Go version 1.23. However, to the best of my knowledge, the latest Go version is 1.21 ([source](https://go.dev/doc/devel/release)). Is it possible that the error is misleading and the actual problem is the patch-level version pin in the `go.mod` file? If so, then moving from `1.21.0` to `1.21` might solve our issue.

@ibarakaiev, can you please add a patch to the recipe that changes to Go version to `1.21`? You can find an example on how to use patches [here](https://github.com/conda-forge/onnxruntime-feedstock/blob/c8396a535f40d6bcffa4e778a3e03833aaa057db/recipe/meta.yaml#L13-L22). If that doesn't help, can you try using the v0.89.0 version of `syft` for this recipe? Thanks!

File: recipes/syft/meta.yaml
Comment (janjagusch) [inline recipes/syft/meta.yaml:21]: I don't think these commands apply to `syft`.

```suggestion
```

File: recipes/syft/meta.yaml
Comment (janjagusch) [inline recipes/syft/meta.yaml:41]: The go-licenses command fails because of an unknown license in one of its dependencies. Let's deal with that later.

```suggestion
    # Clear out cache to avoid file not removable warnings
    - chmod -R u+w $(go env GOPATH) && rm -r $(go env GOPATH)  # [unix]

requirements:
  build:
    - {{ compiler('go-nocgo') }}
    - go-licenses

test:
  commands:
    - syft --version

about:
  home: https://github.com/anchore/syft
  license: Apache-2.0
  license_family: Apache
  license_file:
    - src/LICENSE
```

Comment (xhochy) [reply recipes/syft/meta.yaml]: We won't merge though without the licenses.

**Comment by xhochy** [comment]
> syft requires Go 1.21.0([source](https://github.com/anchore/syft/blob/1067dc2ce9304c8c5b094a6e1b4ad3beabb91746/go.mod#L3)), but that version is not available on CF yet ([source](https://prefix.dev/channels/conda-forge/packages/go-nocgo)). How should we proceed here? Wait for a new Go release on CF? Patch the go.mod file? Package an older version, for example, v0.89.0 ([source](https://github.com/anchore/syft/blob/v0.89.0/go.mod))?

I can look into this later today or Monday

File: recipes/syft/meta.yaml
Comment (janjagusch) [reply recipes/syft/meta.yaml]: > We won't merge though without the licenses.

100% agreed. I just want to check if the build succeeds.

**Comment by janjagusch** [comment]
Unfortunately, we get the same error during testing:

```
+ syft --version
/home/conda/staged-recipes/build_artifacts/syft_1696584351353/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/bin/syft: line 1: syntax error near unexpected token `newline'
/home/conda/staged-recipes/build_artifacts/syft_1696584351353/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/bin/syft: line 1: `!<arch>'
```

Something is wrong with the binary we're trying to compile.

**Comment by xhochy** [comment]
Go 1.21 should be ready now

**Comment by janjagusch** [comment]
> Go 1.21 should be ready now

@ibarakaiev, could you please give this another try?

**Pull request merged**
